src/librustc/query/mod.rs

```diff
@@ -61,7 +61,9 @@ rustc_queries! {
         /// predicate gets in the way of some checks, which are intended
         /// to operate over only the actual where-clauses written by the
         /// user.)
-        query predicates_of(_: DefId) -> Lrc<ty::GenericPredicates<'tcx>> {}
+        query predicates_of(key: DefId) -> Lrc<ty::GenericPredicates<'tcx>> {
+            cache { key.is_local() }
```

Comment:
Would you mind renaming the `cache` modifier to `cache_on_disk_if`? That would make it easier to understand, especially if one is not very familiar with the setup.

Reply:
Done.